exetera/core/dataframe.py

```diff
@@ -29,15 +34,20 @@ def __init__(self, name, dataset,data=None,h5group:h5py.Group=None):
                         h5group<-group-dataset structure, the group has a 'fieldtype' attribute
                          and the dataset is named 'values'.
         """
+        # TODO: consider columns as a name rather than fields
         self.fields = dict()
         self.name = name
         self.dataset = dataset
-        if isinstance(dataset,dst.HDF5Dataset):
-            dataset[name]=self
-        if data is not None:
-            if isinstance(data,dict) and isinstance(list(data.items())[0][0],str) and isinstance(list(data.items())[0][1], fld.Field) :
-                self.fields=data
-        elif h5group is not None and isinstance(h5group,h5py.Group):
+        if isinstance(dataset, dst.HDF5Dataset):
+            dataset[name] = self
+        if dataframe is not None:
+            if isinstance(dataframe, dict) and isinstance(list(dataframe.items())[0][0], str) and\
+                    isinstance(list(dataframe.items())[0][1], fld.Field):
+                self.fields = dataframe
+            else:
+                raise ValueError("if set, 'dataframe' must be a dictionary mapping strings to fields")
+
+        elif h5group is not None and isinstance(h5group, h5py.Group):
             fieldtype_map = {
```

Comment:
The newly constructed dataframe has API for adding fields, so maybe we can use that